sympy/printing/printer.py

```diff
@@ -325,9 +325,10 @@ def _print(self, expr, **kwargs):
                     c.__name__ == classes[0].__name__ or \
                     c.__name__.endswith("Base")) + classes[i:]
             for cls in classes:
-                printmethod = '_print_' + cls.__name__
-                if hasattr(self, printmethod):
-                    return getattr(self, printmethod)(expr, **kwargs)
+                printmethodname = '_print_' + cls.__name__
+                printmethod = getattr(self, printmethodname, None)
+                if printmethod is not None:
+                    return printmethod(expr, **kwargs)
```

Comment:
This change makes it possible to do:
```
_print_Heaviside = None
```
in the CodePrinter class so that all subclasses will ignore the parent `StrPrinter._print_Heaviside` method.

Reply:
Wow. That was the knowledge I was missing when I first worked on this.